### PR TITLE
[Gluten-857] Compilation on GCC8.x and on Debian

### PR DESCRIPTION
**Pull request by glipner**
## What changes were proposed in this pull request?

GCC < 9 needs to have `libstdc++fs` explicitly linked for some of the functions used by Gluten JNI.
Also, when detecting OS in `build_velox.sh`, treat `Debian` same as `Ubuntu`

## How was this patch tested?

Tested to produce working SO files on Dataproc Debian 10 with GCC 8.3.0

### CODE REVIEW

File: cpp/core/CMakeLists.txt [hunk 1]
```diff
@@ -320,6 +334,7 @@ endif()
 target_link_libraries(spark_columnar_jni
     PUBLIC Arrow::arrow Arrow::parquet)
 
+
```
Comment (zhejiangxiaomai) [inline cpp/core/CMakeLists.txt:337]: Please remove extra spaces.

Comment (glipner) [reply cpp/core/CMakeLists.txt]: done

File: cpp/core/CMakeLists.txt [hunk 2]
```diff
@@ -247,6 +247,20 @@ file(MAKE_DIRECTORY ${root_directory}/releases)
 add_library(spark_columnar_jni SHARED ${SPARK_COLUMNAR_PLUGIN_SRCS})
 add_dependencies(spark_columnar_jni jni_proto)
 
+if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
+  execute_process(COMMAND ${CMAKE_C_COMPILER} -print-file-name=libstdc++fs.a
+    RESULT_VARIABLE LIBSTDCXXFS_STATIC_RESULT
+    OUTPUT_VARIABLE LIBSTDCXXFS_STATIC_PATH
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if (LIBSTDCXXFS_STATIC_RESULT EQUAL 0 AND EXISTS "${LIBSTDCXXFS_STATIC_PATH}")
+	  message(STATUS "libstdc++fs.a found at: ${LIBSTDCXXFS_STATIC_PATH}")
```
[the remark this answers is not in the document]
Comment (glipner) [reply cpp/core/CMakeLists.txt]: converted tabs to spaces

File: cpp/core/CMakeLists.txt [hunk 3]
```diff
@@ -247,6 +247,20 @@ file(MAKE_DIRECTORY ${root_directory}/releases)
...
+    target_link_libraries(spark_columnar_jni PRIVATE ${LIBSTDCXXFS_STATIC_PATH})
+  else()
+	  find_library(LIBSTDCXXFS stdc++fs REQUIRED)
```
[the remark this answers is not in the document]
Comment (glipner) [reply cpp/core/CMakeLists.txt]: converted tabs to spaces

**Review by asvyatet — APPROVED** [review]
Cool! 
My privates to Meni

Alex Svyatets

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
@glipner can you resolve conflicts?

**Comment by glipner** [comment]
Conflicts resolved

**Comment by glipner** [comment]
Thanks @FelixYBW !

**Pull request merged**